pkg/ipcache/ipcache.go

```diff
@@ -244,13 +244,21 @@ func (ipc *IPCache) getK8sMetadata(ip string) *K8sMetadata {
 	return nil
 }
 
-// updateNamedPorts accumulates named ports from all K8sMetadata entries to a single map
-func (ipc *IPCache) updateNamedPorts() (namedPortsChanged bool) {
-	if !ipc.needNamedPorts {
+// updateNamedPortsRLocked accumulates named ports from all K8sMetadata entries to a single map
```

Comment:
This should probably explicitly call out that it requires at least a read-lock? Yes the name of the function is clear, but given that the usage of the function is not always called with only the read-lock, then explaining that this call should only be done with _at least_ a read-lock and thus, is safe to be done with a write-lock as well.